src/lib/storage/partitioning/hash_partition_schema.cpp

```diff
@@ -0,0 +1,69 @@
+#include <storage/partitioning/hash_partition_schema.hpp>
+#include <storage/table.hpp>
+
+namespace opossum {
+
+HashPartitionSchema::HashPartitionSchema(ColumnID column_id, HashFunction hash_function,
```

Comment:
if you WERE to inherit from HashFunction, you'd just copy the base class when passing it by value. Use unique/shared_ptr